### PR TITLE
feat: test targets

### PR DESCRIPTION
**Pull request by joske**
## Motivation

Add feature flag to be able to lower coinbase and proof targets for testing purposes. This feature flag will lower the proof and coinbase targets to 8 and 32 respectively.

The feature flag needed to be cascaded to all dependencies of snarkVM to be sure everything uses same constants.

To use, compile with `--features test_targets`
e.g.
```
cargo install --locked --features test_targets --path .
```

## Test Plan

tested by using tx-cannon and adding println to verify correct targets.

[CI link](https://app.circleci.com/pipelines/github/ProvableHQ/snarkOS/13686/workflows/2efc80fc-9c83-47cf-88c0-67b27ee9ab53)

## Related PRs

Must first merge https://github.com/AleoNet/snarkVM/pull/2529 and the snarkVM revision must be updated in snarkOS in 2 places before merging this.

### CODE REVIEW

**Review by vicsn — DISMISSED** [review]
LGTM if you update the Cargo.toml to the rev of https://github.com/AleoNet/snarkVM/pull/2529 (which is just for illustrative purposes, as the maintainer will have to update it anyway)

**Comment by raychu86** [comment]
Does this apply to compiled `snarkOS` binaries via `cargo install --path .`? What does that process look like? Additional documentation in the PR description/in the code would be helpful to know the implications.

**Comment by joske** [comment]
@raychu86 I've updated the description. Let me know if still not clear.

**Comment by raychu86** [comment]
So if `cargo install --locked --features test_targets --path .` is used, the `snarkOS` binary will always have the `test_targets` version of the coinbase targets? We should have some failsafes where the node fails to boot-up in production mode if that is the case. 

We need some sort of enforcement that the `test_targets` is being used only for devnets via `--dev`

File: Cargo.toml
Comment (howardwu) [inline Cargo.toml:48]: **Warning**: snarkOS will no longer be using `AleoNet`'s implementation of snarkVM after this PR.

**Proceed with caution.**

Comment (zkxuerb) [reply Cargo.toml:48]: @joske I'm guessing this was changed for testing purposes and we can change it back to `AleoNet`'s snarkVM ?

Comment (vicsn) [reply Cargo.toml:48]: And merge AleoNet/staging + run `cargo fmt` 🙏

**Comment by vicsn** [comment]
Superseded by https://github.com/AleoNet/snarkOS/pull/3424 to resolve merge conflict

File: Cargo.toml
Comment (joske) [reply Cargo.toml:48]: Yes, this was because the snarkVM PR needs to be merged first. It was always intended that it be changed to an Aleonet snarkVM repo before merging this.

**Comment by vicsn** [comment]
NOTE: this was not actually merged, must be a bug in Github since the branchname is the same as the superseding PR.

**Pull request merged**
